Shaders/PBR/private/ComputeIrradianceMap.psh

```diff
@@ -57,18 +57,19 @@ float3 IrradianceMap(float3 N)
         // Solid angle of current smple
         float OmegaS = 1.0 / (float(g_NumSamples) * pdf);
         
-        float NumPixels = g_EnvMapWidth * g_EnvMapHeight;
+        // Solid angle of 1 pixel across all cube faces
 #if ENV_MAP_TYPE == ENV_MAP_TYPE_CUBE
-        NumPixels *= 6.0;
+        float OmegaP = 4.0 * PI / (6.0 * g_EnvMapWidth * g_EnvMapHeight);
+#else
+        // Solid angle of 1 pixel on sphere
+        float dPhi   = 2.0 * PI / g_EnvMapWidth;
+        float dTheta = PI / g_EnvMapHeight;
+        float OmegaP = (1.0 - L.y * L.y) * dTheta * dPhi;
 #endif
-        
-        // Solid angle of 1 pixel across all cube faces
-        float OmegaP = 4.0 * PI / NumPixels;
-        
         // Applying mip bias produces better results, especially for environments maps with
         // very bright spots.
         float MipBias = 1.0;
-        float MipLevel = max(0.5 * log2(OmegaS / OmegaP) + MipBias, 0.0);
+        float MipLevel = max(0.5 * (log2(OmegaS) - log2(OmegaP)) + MipBias, 0.0);
```

Comment:
Why log2(OmegaS) - log2(OmegaP) is better?

Reply:
Division by zero on the poles